packages/shell-api/src/session.ts

```diff
@@ -106,6 +106,8 @@ export default class Session extends ShellApiWithMongoClass {
     assertArgsDefinedType([fn, options], ['function', [undefined, 'object']]);
     // The driver doesn't automatically ensure that fn is an async
     // function/convert its return type to a Promise, so we do that here.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
```

Comment:
Something's up here. I just hacked it to build for now:

```
src/session.ts(109,5): error TS2322: Type 'Document | undefined' is not assignable to type 'ReturnType<T>'.
  Type 'undefined' is not assignable to type 'ReturnType<T>'.
```

Reply:
https://jira.mongodb.org/browse/NODE-4108 – they changed the return type to be more accurate. We should do the same and update the return type of our wrapper to `Document | undefined` as well.

Reply:
OK I updated it now. The rest of the body of that function should remain unchanged, right?

Reply:
Yes, that is also my understanding :)